drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -84,18 +85,22 @@ impl Handler {
                 // That includes the protocol identifier.
                 // If it is not an ignored protocol, then send an error.
                 if overrun {
-                    if !rx_buf.is_empty() {
-                        if Protocol::from(rx_buf[0]) != Protocol::Ignore {
+                    if rx_bytes != 0 {
+                        if Protocol::from(rx_buf.as_slice()[0])
```

Comment:
These could be folded into a single conditional, I think!

(also, run `cargo xtask clippy` and see what it complains about; I think it would have caught this)